pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,316 @@
+      }
+    }
+  };
+  class MyController {
+    @operation('HEAD', '/checkExist', spec)
+    checkExist(name) {
+    }
+  }
+  ```
+  
+### Commonly-used Operations Decorator
+  
+  Syntax: `@get`, `@post`, `@put`, `@patch`, `@del`
+
+  You can call these suger operation decorators as a shortcut of `@operation`, for example:
+  ```ts
+  class MyController {
+    @get('/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+
+  is equivalent to
+
+  ```ts
+  class MyController {
+    @operation('GET', '/greet', spec)
+    greet(name) {
+    }
+  }
+  ```
+  
+  For more usage, refer to [Routing to Controllers](controllers.htm#routing-to-controllers)
+
+### Parameter Decorator
+
+  Syntax: `@param`
+
+  Optional Pattern: `@param.${in}.${type}(${name})`
+
+  `@param` can be applied to method itself or specific parameters, it describes
+  an input parameter of a Controller method.
+
+  For example:
+
+  ```ts
+    class MyController {
+      @get('/')
+      @param(offsetSpec)
+      @param(pageSizeSpec)
+      list(offset?: number, pageSize?: number) {}
+    }
+  ```
+
+  or
+  
+  ```ts
+    class MyController {
+      @get('/')
+      list(
+        @param(offsetSpec) offset?: number,
+        @param(pageSizeSpec) pageSize?: number,
+      ) {}
+    }
+  ```
+  
+  You can also use this pattern to make it quicker to define params: `@param.${in}.${type}(${name})`
+  
+  - in: one of the following values: `query`, `header`, `path`, `formData`, `body`
+  - type: one of the following values:  `string`, `number`, `boolean`, `integer`
+  - name: a `string`, name of the parameter
+
+  So an example would be `@param.query.number('offset')`. 
+  You can find the specific usage in [Writing Controller methods](controller.htm#writing-controller-methods)
+  
+## Dependency Injection
+
+`@inject` is a decorator to annotate method arguments for automatic injection by 
+LoopBack IoC container. `@inject` can only be used on properties or method parameters.
+
+The `@inject` decorator allows you inject dependencies bound to the application context, 
+to the per-request contextor, to any implementation of the [Context](#context) object. 
+You can bind values, class definitions and provider functions to those contexts and 
+then resolve values (or the results of functions that return those values!) in other 
+areas of your code.
+
+```ts
+// application.ts
+import {Application} from '@loopback/core';
+import 'fs-extra';
+class MyApp extends Application {
+  constructor() {
+    super();
+    const app = this;
+    const widgetConf = JSON.parse(fs.readSync('./widget-config.json'));
+    app.bind('config.widget').to(widgetConf);
+  }
+}
+```
+Now that we've bound the 'config.widget' key to our configuration object, we can
+inject it in our WidgetController:
+
+```ts
+// widget-controller.ts
+import {widgetSpec} from '../apispec';
+@api(widgetSpec)
+class WidgetController {
+  constructor(
+    @inject('config.widget') protected widget: JSON
```

Comment:
The type `JSON` is suspicious.

Reply:
>The type JSON is suspicious.

I wouldn't saturate the example with detail that might obscure the essential meaning in lead-chasing.
`@inject('config.widget') protected widgetConfig: any` should do the trick.

Reply:
@kjdelisle @raymondfeng Sorry I am confused, should the type of `widget` be `any` instead of `JSON`? and could you elaborate more about why `JSON` type is suspicious?

Reply:
@jannyHou It doesn't actually exist (in the sense it's used here, anyway). When I wrote that example, I "paraphrased" the type of the object so that it would be immediately clear what that item was supposed to be, but it's technically an invalid type and wouldn't _actually_ work.